test/blackbox-tests/test-cases/github1811.t/run.t

```diff
@@ -1,8 +1,8 @@
 Check that %{targets} is forbidden in preprocessing actions
 
   $ dune build @all
-  File "dune", line 3, characters 37-47:
+  File "dune", line 3, characters 39-47:
   3 |  (preprocess (action (with-stdout-to %{targets} (run cat %{input-file})))))
-                                           ^^^^^^^^^^
-  Error: Preprocessing actions must not have targets.
+                                             ^^^^^^^^
+  Error: You cannot use %{targets} in preprocessing actions.
```

Comment:
Small quib: the new location is slightly worse. The error message mentions `%{targets}`, but the location only include `targets`.

Reply:
Agreed. It's due to a bug in the lexer. I wrote a fix and will submit it as a separate PR.

Reply:
Done in #4205